sources/packages/api/src/services/student-account-applications/student-account-applications.service.ts

```diff
@@ -0,0 +1,197 @@
+      where: { id },
+    });
+  }
+
+  /**
+   * Get the list of all student account applications
+   * waiting to be assessed by the Ministry.
+   * @returns list of pending student account applications.
+   */
+  async getPendingStudentAccountApplications(): Promise<
+    StudentAccountApplication[]
+  > {
+    return this.repo.find({
+      select: {
+        id: true,
+        submittedDate: true,
+        user: { firstName: true, lastName: true },
+      },
+      relations: {
+        user: true,
+      },
+      where: {
+        assessedDate: IsNull(),
+      },
+      order: {
+        submittedDate: "ASC",
+      },
+    });
+  }
+
+  /**
+   * Create a new student account application to be reviewed by
+   * the Ministry to confirm the student's basic BCeID identity.
+   * @param userName user name that uniquely identifies this user.
+   * @param studentProfile information to be assessed by the Ministry.
+   * @returns student account application created id.
+   */
+  async createStudentAccountApplication(
+    userName: string,
+    studentProfile: StudentAccountApplicationCreateModel,
+  ): Promise<StudentAccountApplication> {
+    const newUser = new User();
+    newUser.userName = userName;
+    newUser.email = studentProfile.email;
+    newUser.firstName = studentProfile.firstName;
+    newUser.lastName = studentProfile.lastName;
+    newUser.creator = newUser;
+
+    const newAccountApplication = new StudentAccountApplication();
+    newAccountApplication.user = newUser;
+    newAccountApplication.creator = newUser;
+    newAccountApplication.submittedData = studentProfile;
+    newAccountApplication.submittedDate = new Date();
+    return this.repo.save(newAccountApplication);
+  }
+
+  /**
+   * Approve the student account application associating the user
+   * with a student account. The Ministry can also adjust any student
+   * data that will then be used to create the student account.
+   * @param id student account application id.
+   * @param studentProfile user and student data to be updated.
+   * @param auditUserId user that should be considered the one that is causing the changes.
+   */
+  async approveStudentAccountApplication(
+    id: number,
+    studentProfile: StudentAccountApplicationApprovalModel,
+    auditUserId: number,
+  ) {
+    const accountApplication = await this.repo.findOne({
+      select: { id: true, user: { id: true } },
+      relations: { user: true },
+      where: { id },
+    });
+
+    if (!accountApplication) {
+      throw new CustomNamedError(
+        "Student account application not found.",
+        STUDENT_ACCOUNT_APPLICATION_NOT_FOUND,
+      );
+    }
+
+    const userInfo = {
+      userId: accountApplication.user.id,
+      lastName: studentProfile.lastName,
+      givenNames: studentProfile.firstName,
+      email: studentProfile.email,
+      birthdate: getISODateOnlyString(studentProfile.dateOfBirth),
+      gender: studentProfile.gender,
+    } as UserInfo;
```

Comment:
Any need for "as UserInfo" here ?

Reply:
The `UserInfo` is not provided completely from this method. It can maybe be declared as Partial in `createStudent` method. I can try to make it more explicit.

Reply:
Created a specific model.